jwt/jwt_ed25519.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 Northern.tech AS
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	    http://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+package jwt
+
+import (
+	"crypto/ed25519"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/pkg/errors"
+)
+
+// JWTHandlerEd25519 is an Ed25519-specific JWTHandler
+type JWTHandlerEd25519 struct {
+	privKey         *ed25519.PrivateKey
+	fallbackPrivKey *ed25519.PrivateKey
+}
+
+func NewJWTHandlerEd25519(privKey *ed25519.PrivateKey,
+	fallbackPrivKey *ed25519.PrivateKey) *JWTHandlerEd25519 {
+	return &JWTHandlerEd25519{
+		privKey:         privKey,
+		fallbackPrivKey: fallbackPrivKey,
```

Comment:
Unfortunately this does not allow migrating from RSA to ED25519. I guess we need to move the fallback logic one level up, or make the JWT handler generic.

Reply:
I reimplemented the fallback using multiple jwtHandler.